src/persist-client/src/lib.rs

```diff
@@ -569,6 +592,177 @@ mod tests {
         assert_eq!(listen.read_until(&6).await, expected_events);
     }
 
+    // Appends need to be contiguous for a shard, meaning the lower of an appended batch needs to
+    // match the current shard upper.
+    #[tokio::test]
+    async fn contiguous_append() {
```

Comment:
nit: all these tests are only (or at least primarily) testing writers. to avoid an explosion of lines in this file, let's move them into mod write, please!

Reply:
> nit: all these tests are only (or at least primarily) testing writers. to avoid an explosion of lines in this file, let's move them into mod write, please!

:+1:

Reply:
I'll move the tests once the rest of the review is done, right before merging. To make my fixups easier to eyeball for you.